Description: Jenkins upstream ships with a forked version of
 JmDNS which has a patched in abort method with shuts down
 uncleanly without blocking.
 .
 Unit this feature ships in JmDNS upstream we will live with a
 blocking shutdown in Debian and Ubuntu.
Author: James Page <james.page@ubuntu.com>
Forwarded: not-needed

Index: jenkins/core/src/main/java/hudson/DNSMultiCast.java
===================================================================
--- jenkins.orig/core/src/main/java/hudson/DNSMultiCast.java	2012-05-02 14:34:03.000000000 +0100
+++ jenkins/core/src/main/java/hudson/DNSMultiCast.java	2012-05-02 14:37:53.066974670 +0100
@@ -52,12 +52,12 @@
 
     public void close() {
         if (jmdns!=null) {
-//            try {
-                jmdns.abort();
+            try {
+                jmdns.close();
                 jmdns = null;
-//            } catch (final IOException e) {
-//                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
-//            }
+            } catch (final IOException e) {
+                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
+            }
         }
     }
 
